src/main/java/org/zalando/nakadi/service/subscription/zk/CuratorZkSubscriptionClient.java

```diff
@@ -290,4 +291,23 @@ public void transfer(final String sessionId, final Collection<Partition.Partitio
             throw new NakadiRuntimeException(e);
         }
     }
+
+    @Override
+    public ZkSubscriptionNode getZkSubscriptionNode() {
+        final ZkSubscriptionNode subscriptionNode = new ZkSubscriptionNode();
+        try {
+            runLocked(() -> {
+                subscriptionNode.setPartitions(listPartitions());
+                subscriptionNode.setSessions(listSessions());
+            });
+        } catch (final NakadiRuntimeException nre) {
+            final Exception cause = nre.getException();
+            if (!(cause instanceof KeeperException.NoNodeException)) {
+                throw new NakadiRuntimeException(cause);
+            }
+            log.info("No data about provided subscription {} in ZK", subscriptionId);
```

Comment:
For me it makes sense to return Optional<ZkSubscriptionNode>, return empty() if there is NoNodeException and rethrow in all other cases.

And in `createSubscriptionStat` method the base for result must be received from topicrepository, while ZkSubscriptionNode must be used only as supplementary node to get more information